src/main/java/org/team1540/robot2025/FieldConstants.java

```diff
@@ -0,0 +1,157 @@
+package org.team1540.robot2025;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.geometry.*;
+import edu.wpi.first.math.util.Units;
+import java.util.*;
+
+// NOTE this file is available at:
+// https://github.com/Mechanical-Advantage/RobotCode2025Public/blob/main/src/main/java/org/littletonrobotics/frc2025/FieldConstants.java
+
+/**
+ * Contains various field dimensions and useful reference points. All units are in meters and poses
+ * have a blue alliance origin.
+ */
+public class FieldConstants {
+    public static final double fieldLength = Units.inchesToMeters(690.876);
+    public static final double fieldWidth = Units.inchesToMeters(317);
+    public static final double startingLineX =
+            Units.inchesToMeters(299.438); // Measured from the inside of starting line
+    public static final double algaeDiameter = Units.inchesToMeters(16);
+
+    public static class Processor {
+        public static final Pose2d centerFace =
+                new Pose2d(Units.inchesToMeters(235.726), 0, Rotation2d.fromDegrees(90));
+    }
+
+    public static class Barge {
+        public static final Translation2d farCage =
+                new Translation2d(Units.inchesToMeters(345.428), Units.inchesToMeters(286.779));
+        public static final Translation2d middleCage =
+                new Translation2d(Units.inchesToMeters(345.428), Units.inchesToMeters(242.855));
+        public static final Translation2d closeCage =
+                new Translation2d(Units.inchesToMeters(345.428), Units.inchesToMeters(199.947));
```

Comment:
Far, Middle, and Close are not constant, they depend on where you measure from. Left, Middle, and Right are constant and mirror appropriately